include/seqan/sequence/string_packed.h

```diff
@@ -585,6 +589,76 @@ resize(
     return front(host(me)).i = new_length;
 }
 
+template <typename TValue, typename THostspec, typename TSize, typename TExpand, typename TValue2>
+inline typename Size< String<TValue, Packed<THostspec> > >::Type
+resize(String<TValue, Packed<THostspec> > & me,
+       TSize new_length,
+       TValue2 const & newValue,
+       Tag<TExpand> tag)
+{
+    typedef String<TValue, Packed<THostspec> > TString;
+    typedef PackedTraits_<TString> TTraits;
+    typedef typename Size<TString>::Type TStringSize;
+    typedef typename TTraits::THostValue THostValue;
+
+    TStringSize old_length = length(me);
+    TStringSize old_host_length = length(host(me));
+
+    if (static_cast<TStringSize>(new_length) < length(me))
+        return resize(me, new_length, tag);
+
+    // create WORD
+    THostValue tmp;
+    tmp.i = 0ul;
+    __uint64 ord = ordValue(TValue(newValue));
+    for (unsigned j = 0; j < TTraits::VALUES_PER_HOST_VALUE; ++j)
+        tmp.i |= ord << (j * TTraits::BITS_PER_VALUE);
```

Comment:
believe me, I have sat a long time on this until it worked, if only a little thing is wrong all oft the tests fail and I have manually compared the entire bit-registers with the version from before that goes through the iterator. This is the right way to do it :sunglasses: